test/token/BasicToken.test.js

```diff
@@ -1,33 +1,82 @@
 import assertRevert from '../helpers/assertRevert';
+const BasicToken = artifacts.require('BasicTokenMock');
 
-var BasicTokenMock = artifacts.require('BasicTokenMock');
+contract('StandardToken', function ([_, owner, recipient, anotherAccount]) {
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
 
-contract('BasicToken', function (accounts) {
-  it('should return the correct totalSupply after construction', async function () {
-    let token = await BasicTokenMock.new(accounts[0], 100);
-    let totalSupply = await token.totalSupply();
+  beforeEach(async function () {
+    this.token = await BasicToken.new(owner, 100);
+  });
+
+  describe('total supply', function () {
+    it('returns the total amount of tokens', async function () {
+      const totalSupply = await this.token.totalSupply();
 
-    assert.equal(totalSupply, 100);
+      assert.equal(totalSupply, 100);
+    });
   });
 
-  it('should return correct balances after transfer', async function () {
-    let token = await BasicTokenMock.new(accounts[0], 100);
-    await token.transfer(accounts[1], 100);
+  describe('balanceOf', function () {
+    describe('when the requested account has no tokens', function () {
+      it('returns zero', async function () {
+        const balance = await this.token.balanceOf(anotherAccount);
 
-    let firstAccountBalance = await token.balanceOf(accounts[0]);
-    assert.equal(firstAccountBalance, 0);
+        assert.equal(balance, 0);
+      });
+    });
 
-    let secondAccountBalance = await token.balanceOf(accounts[1]);
-    assert.equal(secondAccountBalance, 100);
-  });
+    describe('when the requested account has some tokens', function () {
+      it('returns the total amount of tokens', async function () {
+        const balance = await this.token.balanceOf(owner);
 
-  it('should throw an error when trying to transfer more than balance', async function () {
-    let token = await BasicTokenMock.new(accounts[0], 100);
-    await assertRevert(token.transfer(accounts[1], 101));
+        assert.equal(balance, 100);
+      });
+    });
   });
 
-  it('should throw an error when trying to transfer to 0x0', async function () {
-    let token = await BasicTokenMock.new(accounts[0], 100);
-    await assertRevert(token.transfer(0x0, 100));
+  describe('transfer', function () {
+    describe('when the recipient is not the zero address', function () {
+      const to = recipient;
+
+      describe('when the sender does not have enough balance', function () {
+        const amount = 101;
+
+        it('reverts', async function () {
+          await assertRevert(this.token.transfer(to, amount, { from: owner }));
+        });
+      });
+
+      describe('when the sender has enough balance', function () {
+        const amount = 100;
+
+        it('transfer the requested amount', async function () {
```

Comment:
should be `transfers` instead of `transfer`

Reply:
nice catch! thx @gertjanleemans